### DISCUSSION TITLE
How do I view the current price of a stock?

### DISCUSSION THREAD

**Discussion by AndrewC243** [Petersoj/alpaca-java#117] [Q&A] [answered]
Sorry if this is a dumb question, I'm a bit of a novice programmer. Basically I'm trying to make a bracket order that automatically decides the stop-loss, stop-limit, and take profit based off percentage increases or decreases from the current price of the stock, but I haven't been able to find a way to see a stock's price in any of the documentation.

Thanks!

**Comment by Petersoj** [comment, marked as answer]
Hey @AndrewC243,
You'll want to take a look at some of the examples provided in the README like the [`StockMarketDataEndpoint`](https://github.com/Petersoj/alpaca-java#stockmarketdataendpoint) (which is used for historical and current stock data) and the [`StockMarketDataWebsocket`](https://github.com/Petersoj/alpaca-java#stockmarketdatawebsocket) (which is used for real-time streamed stock data). [Here](https://javadoc.io/doc/net.jacobpeterson/alpaca-java/latest/net/jacobpeterson/alpaca/rest/endpoint/marketdata/stock/StockMarketDataEndpoint.html) is the `StockMarketDataEndpoint` Javadoc and [here](https://javadoc.io/doc/net.jacobpeterson/alpaca-java/latest/net/jacobpeterson/alpaca/websocket/marketdata/stock/StockMarketDataWebsocket.html) is the `StockMarketDataWebsocket` Javadoc. Take a look at those Javadocs and examine the various methods they offer to get a better understanding of what stock data you can use in your trading application.
Additionally, I strongly encourage you to read through the entire [Alpaca Trading API documentation](https://alpaca.markets/docs/api-references/trading-api/) and the [Alpaca Stock Market Data API documentation](https://alpaca.markets/docs/api-references/market-data-api/stock-pricing-data/) to gain an understanding of all the features Alpaca has to offer.
To answer your question directly, the following code snippet shows how you can get the current/latest price of AAPL:
```
// Print out latest AAPL trade
Trade latestAAPLTrade = alpacaAPI.stockMarketData().getLatestTrade("AAPL").getTrade();
System.out.printf("Latest AAPL Trade: %s\n", latestAAPLTrade);
```
[Here](https://javadoc.io/doc/net.jacobpeterson/alpaca-java/latest/net/jacobpeterson/alpaca/rest/endpoint/orders/OrdersEndpoint.html) is the Javadoc for the `OrdersEndpoint` which is used to request orders. In addition to the bracket market order method signature [here](https://javadoc.io/static/net.jacobpeterson/alpaca-java/9.1.0/net/jacobpeterson/alpaca/rest/endpoint/orders/OrdersEndpoint.html#requestMarketBracketOrder(java.lang.String,java.lang.Integer,net.jacobpeterson.alpaca.model.endpoint.orders.enums.OrderSide,net.jacobpeterson.alpaca.model.endpoint.orders.enums.OrderTimeInForce,java.lang.Double,java.lang.Double,java.lang.Double)), you also might want to check out the trailing stop percent order method signature [here](https://javadoc.io/static/net.jacobpeterson/alpaca-java/9.1.0/net/jacobpeterson/alpaca/rest/endpoint/orders/OrdersEndpoint.html#requestTrailingStopPercentOrder(java.lang.String,java.lang.Integer,net.jacobpeterson.alpaca.model.endpoint.orders.enums.OrderSide,net.jacobpeterson.alpaca.model.endpoint.orders.enums.OrderTimeInForce,java.lang.Double,java.lang.Boolean)).
Let me know if you have any other questions! Don't forget to star this repository :)

**Comment by AndrewC243** [comment]
Thanks a bunch!
